.github/workflows/main.yml

```diff
@@ -5,12 +5,67 @@ on:
   pull_request:
 
 jobs:
+  code_checks:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        code-check: [black, flakehell]
+        python-version: [3.7]
+    steps:
+      - uses: actions/checkout@v1
+
+      - uses: actions/setup-python@v1
+        with:
+          python-version: ${{ matrix.python-version }}
+          architecture: 'x64'
+
+      - uses: actions/cache@v2
+        if: startsWith(runner.os, 'Linux')
+        with:
+          path: ${{ env.pythonLocation }}
+          key: ${{ env.pythonLocation }}-${{ runner.os }}-${{ matrix.python-version }}-${{ hashFiles('**/*requirements.txt') }}
+          restore-keys: |
+            ${{ runner.os }}-${{ matrix.python-version }}-pip-
+
+      - name: Extract branch name  # works for both 'push' and 'pull_request' events
+        shell: bash
+        run: echo "##[set-output name=branch;]$( if [ ${{ github.event_name }} = push ]; then echo ${GITHUB_REF#refs/heads/}; else echo ${{ github.event.pull_request.head.ref }}; fi )"
+        id: extract_branch
+
+      - name: Install test requirements
+        run: |
+          pip install --upgrade pip
+          pip install --upgrade pip-tools
+          pip install -r requirements/test_requirements.txt
+
+      - name: Run code quality checks
```

Comment:
I think is better to reuse code from Makefile in this step: 
```
run: |
  make ${{ matrix.code-check }}-check
```

Reply:
There's different for CI approach and local. Locally we should check for any changes (not event committed), on the other hand on CI we should compare two remote branches.
So it'll be different `diffs` with different info needed to it.
So I split them not to make logic hard to understand and get lost among params.